### ISSUE TITLE
Error parsing mail with inline attachment

### ISSUE THREAD

**Issue by sergeymild** [michaelspiss/imap_client#16]
If i have a mail with inline image right in message body i received parsing error.

request: 
```await currentFolder.fetch(['BODYSTRUCTURE'], messageIds: [3218], uid: true);```

```
Expected */+/TAG, but got [Word ImapWordType.parenOpen] (
#0      ImapCommand.run (package:imap_client/src/imap_command.dart:102:9)
<asynchronous suspension>
#1      ImapEngine.executeCommand (package:imap_client/src/imap_engine.dart:105:46)
<asynchronous suspension>
#2      _ImapCommandable.sendCommand (package:imap_client/src/imap_commandable.dart:39:20)
#3      ImapFolder.fetch (package:imap_client/src/imap_folder.dart:137:11)
<asynchronous suspension>
#4      _MyHomePageState.reassemble (package:flutter_app0/main.dart:80:35)
<asynchronous suspension>
#5      StatefulElement.reassemble (package:flutter/src/widgets/framework.dart:3933:11)
#6      Element.reassemble.<anonymous closure> (package:flutter/src/widgets/framework.dart:2650:13)
#7      SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4960:14)
#8      Element.reassemble (package:flutter/src/widgets/framework.dart:2649:5)
#9      Element.reasse<…>
```

```
flutter: [INFO] imap_client: S: * 139 FETCH (UID 3218 BODYSTRUCTURE ((("TEXT" "HTML" ("CHARSET" "utf-8") NIL NIL "8BIT" 292 6 NIL NIL NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-25 at 8.35.12 AM.png") "<2156881561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 4973752 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06-25 at 8.35.12 AM.png")) NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-24 at 5.05.55 PM.png") "<2156901561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 478480 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06
flutter: [INFO] imap_client: S: * 139 FETCH (UID 3218 BODYSTRUCTURE ((("TEXT" "HTML" ("CHARSET" "utf-8") NIL NIL "8BIT" 292 6 NIL NIL NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-25 at 8.35.12 AM.png") "<2156881561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 4973752 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06-25 at 8.35.12 AM.png")) NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-24 at 5.05.55 PM.png") "<2156901561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 478480 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06
flutter: [INFO] imap_client: S: * 139 FETCH (UID 3218 BODYSTRUCTURE ((("TEXT" "HTML" ("CHARSET" "utf-8") NIL NIL "8BIT" 292 6 NIL NIL NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-25 at 8.35.12 AM.png") "<2156881561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 4973752 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06-25 at 8.35.12 AM.png")) NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-24 at 5.05.55 PM.png") "<2156901561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 478480 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06
flutter: [INFO] imap_client: S: -24 at 5.05.55 PM.png")) NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-25 at 8.23.25 AM.png") "<2156911561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 1727576 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06-25 at 8.23.25 AM.png")) NIL) "RELATED" ("BOUNDARY" "----==--bound.215693.iva5-90b45dad4fcc.qloud-c.yandex.net") NIL NIL) "MIXED" ("BOUNDARY" "----==--bound.215692.iva5-90b45dad4fcc.qloud-c.yandex.net") NIL NIL))

A7 OK Success
flutter: [INFO] imap_client: S: -24 at 5.05.55 PM.png")) NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-25 at 8.23.25 AM.png") "<2156911561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 1727576 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06-25 at 8.23.25 AM.png")) NIL) "RELATED" ("BOUNDARY" "----==--bound.215693.iva5-90b45dad4fcc.qloud-c.yandex.net") NIL NIL) "MIXED" ("BOUNDARY" "----==--bound.215692.iva5-90b45dad4fcc.qloud-c.yandex.net") NIL NIL))

A7 OK Success
flutter: [INFO] imap_client: S: -24 at 5.05.55 PM.png")) NIL)("IMAGE" "PNG" ("NAME" "Screenshot 2019-06-25 at 8.23.25 AM.png") "<2156911561440963@iva5-90b45dad4fcc.qloud-c.yandex.net>" NIL "BASE64" 1727576 NIL ("INLINE" ("FILENAME" "Screenshot 2019-06-25 at 8.23.25 AM.png")) NIL) "RELATED" ("BOUNDARY" "----==--bound.215693.iva5-90b45dad4fcc.qloud-c.yandex.net") NIL NIL) "MIXED" ("BOUNDARY" "----==--bound.215692.iva5-90b45dad4fcc.qloud-c.yandex.net") NIL NIL))

A7 OK Success
```

**Comment by michaelspiss** [comment]
I finally had time to reproduce the error. It seems like there is a problem with `ImapFolder`'s `_processBody()` method. Fixing this might take a while, sorry. If you find the bug and open a pull request I'd be happy to accept it, but otherwise you might need to wait a while until I can get it fixed.

**Comment by michaelspiss** [comment]
The body/bodystructure parser will get an overhaul before the stable release (partly because of #12), but I cannot give you an estimate regarding when it will happen.

**Comment by michaelspiss** [comment]
Fixed by #19, please contact me if the issue persists
